packages/headless/src/features/product-listing/product-listing-analytics.ts

```diff
@@ -20,89 +19,91 @@ export const logProductListing = (): ProductListingAction =>
     (getState) => new ProductListingAnalyticsProvider(getState)
   );
 
-export const logDocumentOpen = (
+export const logProductRecommendationOpen = (
   productRec: ProductRecommendation
-): ClickAction =>
+): ProductListingAction<AnalyticsType.Click> =>
   makeAnalyticsAction(
     'analytics/productListing/open',
     AnalyticsType.Click,
     (client, state) => {
-      validateProductRecommendationPayload(productRec);
+      validateResultPayload(productRec);
       return client.makeDocumentOpen(
-        partialProductRecommendationInformation(productRec, state),
-        {
-          contentIDKey: 'permanentid',
-          contentIDValue: productRec.permanentid,
-        }
+        partialRecommendationInformation(productRec, state),
+        documentIdentifier(productRec)
       );
-    }
+    },
+    (getState) => new ProductListingAnalyticsProvider(getState)
   );
 
-export const partialProductRecommendationInformation = (
-  productRec: ProductRecommendation,
-  state: Partial<ProductListingAppState>
-): PartialDocumentInformation => {
-  const paginationBasedIndex = (index: number) =>
-    index + (state.pagination?.firstResult ?? 0);
-
-  let productRecIndex = -1;
-
-  const parentResults = state.productListing?.products;
-  productRecIndex = findPositionWithPermanentid(productRec, parentResults);
-
-  if (productRecIndex < 0) {
-    productRecIndex = findPositionInChildResults(productRec, parentResults);
+const documentIdentifier = (
+  productRecommendation: ProductRecommendation
+): DocumentIdentifier => {
+  if (!productRecommendation.permanentid) {
+    console.warn(
+      'Missing field permanentid on productRecommendation. This might cause many issues with your Coveo deployment. See https://docs.coveo.com/en/1913 and https://docs.coveo.com/en/1640 for more information.',
+      productRecommendation
+    );
   }
+  return {
+    contentIDKey: 'permanentid',
+    contentIDValue: productRecommendation.permanentid || '',
```

Comment:
I don't think you need this fallback since you already check for falsey values above

```suggestion
    contentIDValue: productRecommendation.permanentid,
```

Reply:
You're right thanks !